tests/transformations/loop_detection_test.py

```diff
@@ -148,17 +158,16 @@ def test_loop_llvm_canonical_with_extras(with_bounds_check):
     sdfg.add_edge(loopexit, exitstate, dace.InterstateEdge())
 
     xform = CountLoops()
-    assert sdfg.apply_transformations(xform) == 1
-    itvar, rng, _ = xform.loop_information()
-    assert itvar == 'i'
-    assert rng == (0, dace.symbol('N') - 1, 1)
+    assert sdfg.apply_transformations(xform) == 0
```

Comment:
this is very weird because they are loops. We should discuss.
I wonder why this is not detected as a loop because the exiting state (`latch->increment1`) should have the branch in this case and the latch itself is `increment2`. This should count as a `separate_latch=True` case.

Reply:
This should not be a problem, I agree. I explicitly made them _not_ be detected based on the fact that these tests were commented out with the comment above (i.e., that they should not be detected). This made me assume that it would not be desired behavior to detect them as loops. I will see that they can be detected too, but it may require a separate PR.

Reply:
Still strange, _maybe_ ok. We will see as it directly affects my frontend (my states cannot be fused afaik). Is it because of the graph patterns?